tests/FakeItEasy.Analyzer.VisualBasic.Tests/NonVirtualSetupAnalyzerTests.cs

```diff
@@ -0,0 +1,378 @@
+﻿namespace FakeItEasy.Analyzer.VisualBasic.Tests
+{
+    using System.Diagnostics.CodeAnalysis;
+    using FakeItEasy.Analyzer.Tests.Helpers;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Xunit;
+
+    public class NonVirtualSetupAnalyzerTests : DiagnosticVerifier
+    {
+        [Fact]
+        public void Diagnostic_Should_Not_Be_Triggered_For_Interface()
+        {
+            const string Test = @"Imports FakeItEasy
+Namespace TheNamespace
+    Class TheClass
+        Sub Test()
+            Dim foo = A.Fake(Of IFoo)()
+            A.CallTo(Function() foo.Bar()).Returns(42)
+        End Sub
+    End Class
+
+    Interface IFoo
+        Function Bar() As Integer
+    End Interface
+End Namespace
+";
+
+            this.VerifyVisualBasicDiagnostic(Test);
+        }
+
+        [Fact]
+        public void Diagnostic_Should_Not_Be_Triggered_For_Virtual_Member()

```

Comment:
I'm content with either snake case or Pascal. I had noticed the combo, but figured I'd hold off commenting because you were just keeping the same convention that was in use, and that as has been said twice already, maybe another PR would be a better place to fix.